core-web/libs/edit-content/src/lib/components/dot-edit-content-form/dot-edit-content-form.component.html

```diff
@@ -1,30 +1,52 @@
-                        <ng-container *ngTemplateOutlet="tabTemplate; context: { tab: tab }" />
-                    </p-tabPanel>
-                }
-            </p-tabView>
-        } @else {
-            <ng-container *ngTemplateOutlet="tabTemplate; context: { tab: tabs[0] }" />
-        }
-        <ng-template #tabTemplate let-tab="tab">
-            <div class="form__layout">
+    <form [formGroup]="form" class="p-fluid dot-edit-content-form h-100">
+        <p-tabView
+            [scrollable]="true"
+            data-testId="edit-content-multiple-tabs"
+            [class.dot-edit-content-tabview--single-tab]="$hasSingleTab()"
+            class="dot-edit-content-tabview">
+            <!-- Add prepend and append areas to the tabs -->
+            <ng-template
+                dotTabViewInsert
+                [dotTabViewPrepend]="prependContent"
+                [dotTabViewAppend]="appendContent"></ng-template>
+
+            <!-- First tab is not lazy -->
+            @if (firstTab()) {
+                <p-tabPanel [header]="firstTab()!.title" [@fadeIn]>
+                    <ng-container
+                        *ngTemplateOutlet="tabPanelTemplate; context: { tab: firstTab() }" />
+                </p-tabPanel>
+            }
+
+            <!-- Rest of the tabs are lazy -->
+            @for (tab of restOfTabs(); track tab) {
+                <p-tabPanel [header]="tab.title" [@fadeIn]>
+                    <ng-template pTemplate="content">
+                        <ng-container *ngTemplateOutlet="tabPanelTemplate; context: { tab: tab }" />
+                    </ng-template>
+                </p-tabPanel>
+            }
+        </p-tabView>
+
+        <!-- Template for the tab tabPanel -->
+        <ng-template #tabPanelTemplate let-tab="tab">
+            <div class="dot-edit-content-form__layout">
                 @for (row of tab.layout; track row) {
-                    <div class="row" data-testId="row">
+                    <div class="dot-edit-content-form__layout-row" data-testId="row">
                         @for (column of row.columns; track column) {
-                            <div class="column" data-testId="column">
+                            <div class="dot-edit-content-form__layout-column" data-testId="column">
                                 @for (field of column.fields; track field) {
-                                    @if (!isFilteredType(field)) {
-                                        <dot-edit-content-field
-                                            [contentType]="formData.contentType.variable"
-                                            [contentlet]="formData.contentlet"
-                                            [field]="field"
-                                            data-testId="field" />
-                                    }
+                                    <dot-edit-content-field
+                                        [contentType]="contentType.variable"
```

Comment:
The contentType could be null, so I suggest wrapping dot-edit-content-field inside an if statement.

Reply:
It is unnecessary; `contentType` could never be `null`. This creates the shape of tabs that contain the `layout` property and the final `form` (`FormBuilder`).